tests/scripts/parse_notice.py

```diff
@@ -0,0 +1,277 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+
+import json
+import re
+import sys
+from datetime import datetime
+from typing import Any
+
+import requests
+from requests import Response
+
+
+class NoticeParser:
+    POSSIBLE_LICENSE_FILES: list[str] = [
+        "LICENSE",
+        "LICENSE.txt",
+        "LICENSE.rst",
+        "apache-2.0.LICENSE",
+        "apache-2.0.LICENSE.txt",
+    ]
+    POSSIBLE_METADATA_FILES: list[str] = ["METADATA", "METADATA.txt"]
+
+    def __init__(self, requirement_files: list[str], scanned_json_file: str, cli_mode_argument: str) -> None:
+        self.requirement_files: list[str] = requirement_files
+        self.scanned_json_file: str = scanned_json_file
+        self.processed_packages: dict[str, dict[str, str]] = {}
+        self.required_packages: dict[str, str] = {}
+        self.notice_file_name: str = "NOTICE.txt"
+        self.mode: str = cli_mode_argument
+
+        self.read_requirements()
+
+        scanned_results_data = self.read_content_from_file(self.scanned_json_file)
+
+        if not scanned_results_data:
+            raise ValueError(f"{self.scanned_json_file} is empty")
+
+        self.scanned_results_json: Any = json.loads(scanned_results_data)
+
+        notice_file_content: str = self.read_content_from_file("NOTICE.txt")
+
+        package_pattern = r"(?:Package: ([^\n]+))"
+        existing_packages: list[str] = re.findall(package_pattern, notice_file_content)
+        existing_packages.sort()
+
+        requirements_name_from_file = [requirement for requirement in self.required_packages.keys()]
+        requirements_name_from_file.sort()
+
+        real_requirements_name = [requirement for requirement in self.required_packages.values()]
+        real_requirements_name.sort()
+
+        if self.mode == "--check":
+            if real_requirements_name == existing_packages:
```

Comment:
repeated at https://github.com/elastic/elastic-serverless-forwarder/pull/61/files#diff-68dab1ed5621e764af621e1ab471747e31c7615e6f8cf47039e6509d6fd6336fR68-R71

we can just have it once outside of the if/else on `self.mode`